packages/client/tests/e2e/16418-slow-compilation-big-schema/_steps.ts

```diff
@@ -0,0 +1,27 @@
+import { $ } from 'zx'
+
+import { executeSteps } from '../_utils/executeSteps'
+
+void executeSteps({
+  setup: async () => {
+    process.env.PRISMA_SKIP_POSTINSTALL_GENERATE = 'true'
+    await $`pnpm install`
+    await $`pnpm prisma generate`
+  },
+  test: async () => {
+    const timeBefore = Date.now()
+    await $`pnpm exec tsc`
+    const timeAfter = Date.now()
+
+    const timeDiff = timeAfter - timeBefore
+    console.log(`timeDiff: ${timeDiff}`)
+
+    if (timeDiff > 1000 * 60) {
+      throw new Error('Compilation took more than 1 minute')
```

Comment:
How much time does it take at the moment?

Reply:
I think it's about 40s on CI. I shared that in the original issue we should probably measure instantiations, and memory instead of time. I hope to get to this some day :smile: